core/unit_test/TestViewPadding.hpp

```diff
@@ -0,0 +1,618 @@
+}
+
+template <typename dim_t, typename viewDim, typename layout,
+          typename allow_padding, typename... Vals>
+typename std::enable_if_t<std::is_same<allow_padding, std::true_type>::value,
+                          bool>
+padding_construction(int rank, Vals... params) {
+  using view_t      = Kokkos::View<dim_t, layout>;
+  using offset_t    = Kokkos::Impl::ViewOffset<viewDim, layout, void>;
+  using use_padding = std::bool_constant<
+      allow_padding() &&
+      !std::is_same<layout, typename Kokkos::LayoutStride>::value>;
+  using padding =
+      std::integral_constant<unsigned int,
+                             use_padding::value ? sizeof(DType) : 0>;
+  offset_t offset = offset_t(padding(), layout(params...));
+  auto alloc_prop =
+      Kokkos::view_alloc("vDim" + std::to_string(rank), Kokkos::AllowPadding);
+
+  view_t v(alloc_prop, params...);
+  return ASSERT_VIEW_OFFSET_STRIDES(v, offset, rank) ? true : false;
+}
+
+template <typename dim_t, typename viewDim, typename layout,
+          typename allow_padding, typename... Vals>
+typename std::enable_if_t<std::is_same<allow_padding, std::false_type>::value,
+                          bool>
+padding_construction(int rank, Vals... params) {
+  using view_t      = Kokkos::View<dim_t, layout>;
+  using offset_t    = Kokkos::Impl::ViewOffset<viewDim, layout, void>;
+  using use_padding = allow_padding;
+  using padding =
+      std::integral_constant<unsigned int,
+                             use_padding::value ? sizeof(DType) : 0>;
+  offset_t offset = offset_t(padding(), layout(params...));
+  auto alloc_prop = Kokkos::view_alloc("vDim" + std::to_string(rank));
+  view_t v(alloc_prop, params...);
+  return ASSERT_VIEW_OFFSET_STRIDES(v, offset, rank);
+}
+
+template <typename view_t, typename... Vals>
+bool padding_assignment_and_copy(int rank, Vals... params) {
+  bool result = true;
+
+  auto alloc_prop =
+      Kokkos::view_alloc("vDim" + std::to_string(rank), Kokkos::AllowPadding);
+  view_t v(alloc_prop, params...);
+  deep_copy(v, 1);
+  auto v_mirror = Kokkos::create_mirror_view(v);
+  result &= ASSERT_VIEW_STRIDES(v_mirror, v, rank);
+
+  /*deep_copy should support strided views if stride from padding*/
+  Kokkos::deep_copy(v_mirror, v);
+  return result;
+}
+template <typename dim_t, typename layout, typename viewDim>
+auto get_subview(
+    Kokkos::View<dim_t, layout> &view,
+    [[maybe_unused]] typename std::enable_if_t<
+        std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_0>>::value, void>
+        *ptr = nullptr) {
+  return Kokkos::subview(view, 0);
+}
+
+template <typename dim_t, typename layout, typename viewDim>
+auto get_subview(
+    Kokkos::View<dim_t, layout> &view,
+    [[maybe_unused]] typename std::enable_if_t<
+        std::is_same<viewDim, Kokkos::Impl::ViewDimension<ARG_1>>::value, void>
+        *ptr = nullptr) {
+  auto pair = pair_t(SUBRANGE_DIM);
+  return Kokkos::subview(view, pair);
+}
```

Comment:
The `get_subview` overloads can almost certainly be consolidated down to one case. If nothing else, the body can just be a ladder of `if constexpr` on the same conditions, but I think it can actually be written generically. I'll only push on going at least as far as `if constexpr` though.

Reply:
No, I think that wouldn't work as the return value type is different.